test/v2/controllers/V2DeparturesControllerSpec.scala

```diff
@@ -858,6 +862,85 @@ class V2DeparturesControllerSpec
 
   }
 
+  "retrieving a single departure/movement" - {
+    "should return ok with json body of departure" in {
+      val createdTime = OffsetDateTime.now()
+      val departureResponse = DepartureResponse(
+        DepartureId("0123456789abcdef"),
+        EORINumber("GB123"),
+        EORINumber("XI456"),
+        Some(MovementReferenceNumber("312")),
+        createdTime,
+        createdTime
+      )
+
+      when(mockDeparturesPersistenceService.getDeparture(EORINumber(any()), DepartureId(any()))(any(), any()))
+        .thenAnswer(
+          _ => EitherT.rightT(departureResponse)
+        )
+
+      val request = FakeRequest(
+        "GET",
+        routing.routes.DeparturesRouter.getDeparture("123").url,
+        headers = FakeHeaders(Seq(HeaderNames.ACCEPT -> "application/vnd.hmrc.2.0+json")),
+        AnyContentAsEmpty
+      )
+      val result = sut.getDeparture(DepartureId("0123456789abcdef"))(request)
+
+      status(result) mustBe OK
+      contentAsJson(result) mustBe Json.toJson(
+        HateoasDepartureResponse(
+          DepartureId("0123456789abcdef"),
+          DepartureResponse(
+            DepartureId("0123456789abcdef"),
+            EORINumber("GB123"),
+            EORINumber("XI456"),
+            Some(MovementReferenceNumber("312")),
```

Comment:
As are these

Reply:
Ok thanks for letting me know. Just so we don’t need to go back and forth, can you please clarify the scope of your ask? As I mentioned earlier, magic strings are everywhere. Have just been following other example code, so I am guessing we need to create some tech debt stories to eliminate magic strings?

Is there already/or is it worth writing a standards doc to simplify the PR process?

Thanks.

Reply:
I think his point was more that we should open a tech debt ticket to consider replacing the use of these hard coded values in the test with generators so that we're testing arbitary values rather than a specific value. You're right in the fact it's being done everywhere and it would be ludicrous to ask you to go change everything in this PR.

For example, [in some of the integration tests, we've created model generators that generate random numbers/strings each time.](https://github.com/hmrc/common-transit-convention-traders/blob/9cacbd8e085d5a79d1458ed28a792d828fe9e3dc/it/v2/connectors/PersistenceConnectorSpec.scala#L87-L115). It just heightens the chance we'll catch something weird in edge cases.

I'd definately argue that there is a case for them, but equally, I'd argue that there are times where you just need a static valued test - it all boils down to judgement and style, really. I suspect that this is one of those times where it could benefit.

If you fancy giving that a go with your stuff, great, but don't worry about doing it with anything that is outside the scope of your ticket.

Reply:
Oh, so the ask is not about magic strings, more around not using fixed values in tests - so using dynamically generated values/property based testing? Ok, will take a look at the examples - thanks for the explanation :)